provider/cluster_waiter_resource.go

```diff
@@ -81,40 +83,12 @@ func (r *ClusterWaiterResource) Create(ctx context.Context,
 		return
 	}
 
-	state.Ready = types.Bool{
-		Value: false,
-	}
-
-	timeout := defaultTimeoutInMinutes
-	if !state.Timeout.Unknown && !state.Timeout.Null {
-		if state.Timeout.Value <= 0 {
-			response.Diagnostics.AddWarning(nonPositiveTimeoutSummary, fmt.Sprintf(nonPositiveTimeoutFormat, state.Cluster.Value))
-		} else {
-			timeout = state.Timeout.Value
-		}
-	}
+	state, err := r.startPolling(ctx, state)
 
-	// Wait till the cluster is ready:
-	object, err := r.retryClusterReadiness(3, 30*time.Second, state.Cluster.Value, ctx, timeout)
 	if err != nil {
-
-		response.Diagnostics.AddError(
-			"Can't poll cluster state",
-			fmt.Sprintf(
-				"Can't poll state of cluster with identifier '%s': %v",
-				state.Cluster.Value, err,
-			),
-		)
+		response.Diagnostics.AddError("Can't poll cluster state", err.Error())
```

Comment:
Please add here information if this is during create or update
Something like:
```suggestion
		response.Diagnostics.AddError("Can't poll cluster state (Create resource)", err.Error())
```

Reply:
done